src/Features/CSharp/Portable/Copilot/CSharpImplementNotImplementedExceptionFixProvider.cs

```diff
@@ -0,0 +1,172 @@
+            CSharpFeaturesResources.Warning_colon_AI_suggestions_might_be_inaccurate);
+
+    public override ImmutableArray<string> FixableDiagnosticIds { get; }
+        = [IDEDiagnosticIds.CopilotImplementNotImplementedExceptionDiagnosticId];
+
+    public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        if (context.Diagnostics.Length == 0)
+        {
+            return;
+        }
+
+        var document = context.Document;
+        var cancellationToken = context.CancellationToken;
+
+        if (document.GetLanguageService<ICopilotOptionsService>() is not { } optionsService ||
+            await optionsService.IsImplementNotImplementedExceptionEnabledAsync().ConfigureAwait(false) is false)
+        {
+            return;
+        }
+
+        if (document.GetLanguageService<ICopilotCodeAnalysisService>() is not { } copilotService ||
+            await copilotService.IsAvailableAsync(cancellationToken).ConfigureAwait(false) is false)
+        {
+            return;
+        }
+
+        var throwNode = context.Diagnostics[0].Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+
+        // Preliminary analysis before registering fix
+        var methodOrProperty = throwNode.FirstAncestorOrSelf<MemberDeclarationSyntax>();
+        if (methodOrProperty is BasePropertyDeclarationSyntax or BaseMethodDeclarationSyntax)
+        {
+            var fix = DocumentChangeAction.New(
+                title: CSharpAnalyzersResources.Implement_with_Copilot,
+                createChangedDocument: async (_, cancellationToken) => await GetDocumentUpdater(context: context, diagnostic: null)(cancellationToken).ConfigureAwait(false),
+                createChangedDocumentPreview: (_, _) => Task.FromResult(context.Document),
+                equivalenceKey: nameof(CSharpAnalyzersResources.Implement_with_Copilot));
+            context.RegisterCodeFix(fix, context.Diagnostics[0]);
+            Logger.Log(FunctionId.Copilot_Implement_NotImplementedException_Fix_Registered, logLevel: LogLevel.Information);
+        }
+    }
+
+    protected override async Task FixAllAsync(
+        Document document, ImmutableArray<Diagnostic> diagnostics,
+        SyntaxEditor editor, CancellationToken cancellationToken)
+    {
+        var memberReferencesBuilder = ImmutableDictionary.CreateBuilder<SyntaxNode, ImmutableArray<ReferencedSymbol>>();
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+        foreach (var diagnostic in diagnostics)
+        {
+            var throwNode = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+            var methodOrProperty = throwNode.FirstAncestorOrSelf<MemberDeclarationSyntax>();
+
+            Contract.ThrowIfNull(methodOrProperty);
+            Contract.ThrowIfFalse(methodOrProperty is BasePropertyDeclarationSyntax or BaseMethodDeclarationSyntax);
```

Comment:
FWIW, this `ThrowIfFalse implies that methodOrProperty isn't null, so the `ThrowIfNull` on line 89 is redundant.